File: static/code/sample/tethered/gdl-sertc-b.py
# In this sample:
# Set local date/time and read every second.)

import time
from datetime import datetime
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)


# method
def SetDateTime(dt):
    year = dt.year
    month = dt.month
    day = dt.day
    hour = dt.hour
    minute = dt.minute
    second = dt.second

    duelink.Engine.ExecuteCommand(
        f"SetTime({year}, {month},{day},{hour},{minute},{second})"
    )

def GetDateTime():
    duelink.Engine.ExecuteCommand("ReadTime()")

    year = int(duelink.Engine.ExecuteCommand("_y"))
    month = int(duelink.Engine.ExecuteCommand("_n"))
    day = int(duelink.Engine.ExecuteCommand("_d"))
    hour = int(duelink.Engine.ExecuteCommand("_h"))
    minute = int(duelink.Engine.ExecuteCommand("_m"))
    second = int(duelink.Engine.ExecuteCommand("_s"))

    return datetime(year, month, day, hour, minute, second)

SetDateTime(datetime.now())

while True:
    print(f"Now: {GetDateTime()} ")
    time.sleep(1)  # 1000 ms

File: static/code/sample/tethered/gdl-sertc-b.cs
// In this sample:
// Set local date/time and read every second.

using System.Data.SqlTypes;
using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

void SetDateTime(DateTime dt) {
    var year = dt.Year;
    var month = dt.Month;
    var day = dt.Day;
    var hour = dt.Hour;
    var minute = dt.Minute;
    var second = dt.Second;

    duelink.Engine.ExecuteCommand($"SetTime({year}, {month},{day},{hour},{minute},{second})");
}

DateTime GetDateTime() {
    duelink.Engine.ExecuteCommand("ReadTime()");

    var year = (int)duelink.Engine.ExecuteCommand("_y"); ;
    var month = (int)duelink.Engine.ExecuteCommand("_n"); ;
    var day = (int)duelink.Engine.ExecuteCommand("_d"); ;
    var hour = (int)duelink.Engine.ExecuteCommand("_h"); ;
    var minute = (int)duelink.Engine.ExecuteCommand("_m"); ;
    var second = (int)duelink.Engine.ExecuteCommand("_s"); ;

    return new DateTime(year,month,day,hour,minute,second);
}

SetDateTime(DateTime.Now);

while (true) {   
    Console.WriteLine($"Now: {GetDateTime().ToString()} ");    

    Thread.Sleep(1000);
}


